.github/workflows/scripts/css_check.py

```
@@ -0,0 +1,350 @@
+        file_path: Path to the TypeScript file to process.
+        directory: Base directory being scanned.
+        allowed_css_patterns: List of allowed CSS file patterns.
+        violations: List to store CSS violations.
+        correct_css_imports: List to store correct CSS imports.
+        embedded_css_violations: List to store embedded CSS violations.
+
+    Returns:
+        None: This function modifies provided lists & does not return any value.
+    """
+    try:
+        with open(file_path, "r", encoding="utf-8") as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)